Update aioairzone to v0.1.0

Noltari opened the pull request:
## Proposed change
Update aioairzone to v0.1.0 and add code improvements needed for https://github.com/home-assistant/core/pull/67924 and https://github.com/home-assistant/core/pull/68140.

Release notes:
- https://github.com/Noltari/aioairzone/releases/tag/0.0.3
- https://github.com/Noltari/aioairzone/releases/tag/0.0.4
- https://github.com/Noltari/aioairzone/releases/tag/0.1.0

Git compare: https://github.com/Noltari/aioairzone/compare/0.0.2...0.1.0

## Type of change
- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description. https://github.com/Noltari/aioairzone/compare/0.0.2...0.1.0
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

Noltari committed:
Update aioairzone to 0.1.0

frenck commented on homeassistant/components/airzone/__init__.py:
```diff
@@ -26,17 +26,20 @@ def __init__(
         entry: ConfigEntry,
         system_zone_id: str,
-        zone_name: str,
+        zone_data: dict,
```
Please make the dictionary typing more specific

Noltari replied:
Fixed in 66bd22912bbcf6577569050a66093a5dd4394fab

frenck commented on homeassistant/components/airzone/__init__.py:
```diff
@@ -26,17 +26,20 @@ def __init__(
-            "name": f"Airzone [{system_zone_id}] {zone_name}",
+            "name": f"Airzone [{system_zone_id}] {zone_data[AZD_NAME]}",
         }
+        self.airzone: AirzoneLocalApi = coordinator.airzone
```
We don't need to store this? The coordinator is already available everywhere, as this class extends the CoordinatorEntity

Noltari replied:
Removed in 66bd22912bbcf6577569050a66093a5dd4394fab

frenck commented on homeassistant/components/airzone/coordinator.py:
```diff
@@ -23,7 +23,7 @@ class AirzoneUpdateCoordinator(DataUpdateCoordinator):
     def __init__(self, hass: HomeAssistant, airzone: AirzoneLocalApi) -> None:
         """Initialize."""
-        self.airzone = airzone
+        self.airzone: AirzoneLocalApi = airzone
```
Isn't this already inferred?

Noltari replied:
Reverted change in 66bd22912bbcf6577569050a66093a5dd4394fab

frenck commented on homeassistant/components/airzone/sensor.py:
```diff
@@ -76,11 +74,11 @@ def __init__(
         entry: ConfigEntry,
         system_zone_id: str,
-        zone_name: str,
+        zone_data: dict,
```
Please make the dict more specific

Noltari replied:
Fixed in 66bd22912bbcf6577569050a66093a5dd4394fab

Noltari committed:
airzone: add changes suggested by @frenck

- Make dicts more specific.
- Remove inferred type
- Remove airzone from AirzoneEntity.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari commented:
Many thanks for the review @frenck

frenck reviewed (approved):
Thanks, @Noltari 👍
